### PR TITLE
Add file reset functionality for text and JSON files in temp.py

### PR DESCRIPTION
**Pull request by umital-intel**
_The PR review is to check for sustainability and correctness.  Sustainability is actually more business critical as correctness is largely tested into the code over time.   Its useful to keep in mind that SW often outlives the HW it was written for and engineers move from job to job so it is critical that code developed for Intel be supportable across many years.  It is up to the submitter and reviewer to look at the code from a perspective of what if we have to debug this 3 years from now after the author is no longer available and defect databases have been lost.  Yes, that happens all the time when we are working with time scales of more than 2 years.  When reviewing your code it is important to look at it from this perspective._

Author Mandatory (to be filled by PR Author/Submitter)
------------------------------------------------------
- Developer who submits the Pull Request for merge is required to mark the checklist below as applicable for the PR changes submitted.  
- Those checklist items which are not marked are considered as not applicable for the PR change.  
- Items marked with an asterisk suffix are mandatory items to check and if not marked will be treated as non-compliant pull requests by the developers for Inner Source Development Model (ISDM) compliance

### PULL DESCRIPTION
_Provide a 1-2 line brief overview of the changes submitted through the Pull Request..._

### REFERENCES
Reference URL for issue tracking (JIRA/HSD/Github): **\<URL to be filled>**
- [ ] **_Added label to the Pull Request following the template: ISDM\_\<Complexity>\*_** \
	Note-1: Depending on complexity of code changes, use the suitable word for complexity: Low/Medium/High \
	Example: PR for Slim boot loader project with medium complexity can have the label as: ISDM_Medium	
- [ ] Added label to the Pull Request for easier discoverability and search
- [ ] RTC or HSD number will be included in final merge. HSD must always be included if available.
- [ ] Changelogs are updated (or N/A if not customer visible)
- [ ] inbm/log_changes.txt and inbm-vision/log_changes.txt are updated for potentially Validation-breaking log changes (or N/A if none)

### CODE MAINTAINABILITY
- [ ] **_Commit Message meets guidelines as indicated in the URL\*_**
	- (https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md)
- [ ] **_Every commit is a single defect fix and does not mix feature addition or changes\*_**
- [ ] Added required new tests relevant to the changes
	- [ ] PR contains URL links to functional tests executed with the new tests 
- [ ] Updated Documentation as relevant to the changes
- [ ] Updated Build steps/commands changes as relevant
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide description)
- [ ] Specific instructions or information for code reviewers (If any):
- [ ] Run 'go fmt' or format-python.sh as applicable.
- [ ] New/modified methods and functions should have type annotations on signatures as applicable
- [ ] New/modified methods must have appropriate doc strings (language dependent)

### APPLICATION SPECIFIC
- [ ] Does PR change default config files under /etc? If so, will application still work after an upgrade that leaves /etc alone, like a Mender upgrade?
- [ ] Is cloud UI changed? If so, are cloud definition files updated?

Maintainer Mandatory (to be filled by PR Reviewer/Approving Maintainer)
-----------------------------------------------------------------------
- Maintainer who approves the Pull Request for merge is required to mark the checklist below as appropriate for the PR change reviewed as key proof of attestation indicating reasons for merge. 
- Those checklist items which are not marked are considered as not applicable for the PR change. 
- Items marked with an asterisk suffix are mandatory items to check and if not marked will be treated as non-compliant pull requests by the maintainers for ISDM compliance.

### QUALITY CHECKS
- [ ] Architectural and Design Fit
- [ ] **_Quality of code (At least one should be checked as applicable)\*_**
	- [ ] Commit Message meets guidelines
	- [ ] PR changes adhere to industry practices and standards
	- [ ] Error and exception code paths implemented correctly
	- [ ] Code reviewed for domain or language specific anti-patterns
	- [ ] Code is adequately commented
	- [ ] Code copyright is correct
	- [ ] Confusing logic is explained in comments
	- [ ] Commit comment can be used to design a new test case for the changes
- [ ] **_Test coverage shows adequate coverage with required CI functional tests pass on all supported platforms\*_**
- [ ] **_Static code scan report shows zero critical issues\*_**
- [ ] Integration tests are passing

### CODE REVIEW IMPACT
- Summary of Defects Detected in Code Review: **\<%P1*xx,P2*xx,P3*xx,P4*xx%>** \
Note P1/P2/P3/P4 denotes severity of defects found (Showstopper/High/Medium/Low) and xx denotes number of defects found

### SECURITY CHECKS
Please check if your PR fulfills the following requirements:

- [ ] Follow best practices when handling primitive data types
- [ ] Configure minimal permissions when opening pipes and ports
- [ ] Check contents within input structures are valid before use
- [ ] All forms of input validated
- [ ] Avoid inter-process race conditions
- [ ] Error and exception handling implemented
- [ ] Defend against Canonical Representation Issues - Any paths utilized?
- [ ] Follow 'secure by default' - Any configuration values added
- [ ] Fail safe - Any failure scenarios?
- [ ] Clean up temporary files - Any temporary files being used?

# _Code must act as a teacher for future developers_

<h1 style='color: red;'>${\color{red}DO \space NOT \space EDIT \space ANYTHING}$</h1>

>### Start of GenAI Co-Pilot generated PR Summary

## Types of major changes in the pull request
enhancement

___

## Pull request change summary
- Introduced a new function reset_files in temp.py to clear the contents of a file.
  - The function takes filepath and file_type as arguments.
  - Supports clearing text files by writing an empty string.
  - Supports clearing JSON files by writing an empty JSON object.
  - Logs an error if an unsupported file type is provided.

___

## Details of Pull Request changes by File
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>temp.py&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        temp.py<br><br>

**Adds a new function reset_files to reset the content of <br>files based on the file type provided. It supports text and <br>JSON file types and logs an error for invalid types.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/umital-intel/intel-inb-manageability/pull/1/files#diff-beaca8fc85dad9ba862a8fa7fa9cc55730ce6696bc447a0a19ae321f3c1c3829"> +8/-0</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

>### End of Co-Pilot generated PR Summary

### CODE REVIEW

**Comment by nex-maximus** [comment]
## Code Review Analysis

### Release Version: v3.7.1

### (review updated until commit https://github.com/umital-intel/intel-inb-manageability/commit/a22750085e39c3aa2dcb6a6fa717e5259d4c3bbf)


- 🎯 **Main theme:** Adding a function to reset text and JSON files to their default empty states.
- 📝 **PR summary:** The PR introduces a new function reset_files in temp.py that allows for resetting text and JSON files to an empty state. The function takes a file path and an optional file type argument, defaulting to text files, and clears the file contents accordingly.
- 📌 **Type of PR:** Enhancement
- 🏅 **Score:** 75
- 🧪 **Relevant tests added:** No
- ⏱️ **Estimated effort to review [1-5]:** 2, because the PR is small and the changes are straightforward, but it lacks error handling and tests.
- 🔒 **Security concerns:** No
## Code Review Feedback



<strong>💡 General suggestions:</strong> The new reset_files function is a useful addition for managing temporary files. However, it is important to ensure that the function is robust and handles potential errors gracefully. It would also be beneficial to include tests to verify the functionality of reset_files and to ensure that it works as expected with different file types and edge cases.


<details> <summary><strong>✨ Usage tips:</strong></summary><hr> 

> Tag me in a comment '@nex-maximus' and add one of the following commands:
> **/review [-i]**: Request a review of your Pull Request. For an incremental review, which only considers changes since the last review, include the '-i' option.
> **/describe**: Modify the PR title and description based on the contents of the PR.
>For example: /review --pr_reviewer.extra_instructions="focus on the file: ..." 

</details>

File: temp.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+def reset_files(filepath, file_type="txt"):
+    with open(filepath, "w") as output_file:
+        if file_type == "txt":
+            output_file.write("")
+        elif file_type == "json":
+            output_file.write(json.dumps({}))
+        else:
+            log.error("ERROR: Invalid file type")
```
Comment (nex-maximus) [inline temp.py:8]: Consider raising an exception when an unsupported file type is provided. This will make the error more explicit and prevent potential issues down the line. [important]

File: temp.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+def reset_files(filepath, file_type="txt"):
```
Comment (nex-maximus) [inline temp.py:1]: It would be beneficial to add a docstring to the `reset_files` function to explain its purpose, arguments, and behavior. [medium]

**Comment by nex-maximus** [comment]
**[PR Description](https://github.com/umital-intel/intel-inb-manageability/pull/1)** updated to latest commit (https://github.com/umital-intel/intel-inb-manageability/commit/339547a47eaceb9544beccaf2a5dcbb95f82497c)

File: temp.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+    with open(filepath, "w") as output_file:
```
Comment (nex-maximus) [inline temp.py:2]: Consider adding a check to ensure the file exists before trying to open it. This can prevent runtime errors. [important]

File: temp.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+        if file_type == "txt":
```
Comment (nex-maximus) [inline temp.py:3]: The function currently only supports text and JSON files. Consider making it more generic to handle other file types as well. [medium]

**Comment by nex-maximus** [comment]
**[Persistent review](https://github.com/umital-intel/intel-inb-manageability/pull/1#issuecomment-1884461272)** updated to latest commit https://github.com/umital-intel/intel-inb-manageability/commit/339547a47eaceb9544beccaf2a5dcbb95f82497c

File: temp.py (hunk 1 above)
Comment (nex-maximus) [inline temp.py:8]: Consider using Python's logging module instead of a generic `log` object for logging errors. [medium]

**Comment by nex-maximus** [comment]
**[Persistent review](https://github.com/umital-intel/intel-inb-manageability/pull/1#issuecomment-1884461272)** updated to latest commit https://github.com/umital-intel/intel-inb-manageability/commit/339547a47eaceb9544beccaf2a5dcbb95f82497c

File: temp.py (hunk 1 above)
Comment (nex-maximus) [inline temp.py:8]: Consider adding a return statement in the error handling section to ensure the function stops executing after logging the error. [important]

File: temp.py (hunk 3 above)
Comment (nex-maximus) [inline temp.py:2]: The function could be made more robust by checking if the file exists before trying to open it. This could prevent runtime errors. [medium]

File: temp.py (hunk 1 above)
Comment (nex-maximus) [inline temp.py:8]: **Suggestion:** Handle the case where the file at the given filepath does not exist or is not writable. [error handling]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
import os

def reset_files(filepath, file_type="txt"):
    if not os.path.exists(filepath):
        log.error("ERROR: File does not exist")
        return
    if not os.access(filepath, os.W_OK):
        log.error("ERROR: File is not writable")
        return
    with open(filepath, "w") as output_file:
        if file_type == "txt":
            output_file.write("")
        elif file_type == "json":
            output_file.write(json.dumps({}))
        else:
            log.error("ERROR: Invalid file type")
```

Comment (nex-maximus) [inline temp.py:8]: **Suggestion:** Allow the user to specify a custom reset value for the file. [enhancement]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
def reset_files(filepath, file_type="txt", reset_value=""):
    with open(filepath, "w") as output_file:
        if file_type == "txt":
            output_file.write(reset_value)
        elif file_type == "json":
            output_file.write(json.dumps(reset_value) if reset_value else json.dumps({}))
        else:
            log.error("ERROR: Invalid file type")
```

**Comment by nex-maximus** [comment]
**[Persistent review](https://github.com/umital-intel/intel-inb-manageability/pull/1#issuecomment-1884461272)** updated to latest commit https://github.com/umital-intel/intel-inb-manageability/commit/339547a47eaceb9544beccaf2a5dcbb95f82497c

File: temp.py (hunk 2 above)
Comment (nex-maximus) [inline temp.py:1]: Import the `json` and `logging` modules at the beginning of the file. [important]

File: temp.py (hunk 3 above)
Comment (nex-maximus) [inline temp.py:2]: Handle exceptions when opening the file using a try-except block. This will prevent the program from crashing if the file does not exist or cannot be opened. [important]

File: temp.py (hunk 1 above)
Comment (nex-maximus) [inline temp.py:8]: Instead of logging an error when an unsupported file type is provided, consider raising an exception. This will make it clear that an error has occurred and will allow the caller to handle it appropriately. [medium]

Comment (nex-maximus) [inline temp.py:8]: **Suggestion:** Handle the case where the file does not exist or is not accessible. [enhancement]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
import json
import os

def reset_files(filepath, file_type="txt"):
    if not os.path.exists(filepath):
        log.error(f"ERROR: File {filepath} does not exist")
        return
    try:
        with open(filepath, "w") as output_file:
            if file_type == "txt":
                output_file.write("")
            elif file_type == "json":
                output_file.write(json.dumps({}))
            else:
                log.error("ERROR: Invalid file type")
    except PermissionError:
        log.error(f"ERROR: No permission to write to file {filepath}")
```

Comment (nex-maximus) [inline temp.py:8]: **Suggestion:** Import the json module to use json.dumps(). [bug]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
import json

def reset_files(filepath, file_type="txt"):
    with open(filepath, "w") as output_file:
        if file_type == "txt":
            output_file.write("")
        elif file_type == "json":
            output_file.write(json.dumps({}))
        else:
            log.error("ERROR: Invalid file type")
```

File: temp.py [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+            output_file.write("")
+        elif file_type == "json":
+            output_file.write(json.dumps({}))
```
Comment (nex-maximus) [inline temp.py:6]: Import the `json` module at the beginning of the file to ensure that `json.dumps({})` can be used within the `reset_files` function. [important]

**Comment by nex-maximus** [comment]
**[Persistent review](https://github.com/umital-intel/intel-inb-manageability/pull/1#issuecomment-1884461272)** updated to latest commit https://github.com/umital-intel/intel-inb-manageability/commit/a22750085e39c3aa2dcb6a6fa717e5259d4c3bbf

File: temp.py (hunk 1 above)
Comment (nex-maximus) [inline temp.py:8]: Add an `import logging` statement to ensure that `log.error` can be used for logging errors. [important]

Comment (nex-maximus) [inline temp.py:8]: Consider raising an exception instead of logging an error for an invalid file type to allow the calling code to handle the error appropriately. [medium]

File: temp.py (hunk 3 above)
Comment (nex-maximus) [inline temp.py:2]: Add a check to ensure that the `filepath` provided is a valid path and that the file exists before attempting to open it. [medium]

File: temp.py [hunk 6]
```diff
@@ -0,0 +1,8 @@
+def reset_files(filepath, file_type="txt"):
+    with open(filepath, "w") as output_file:
+        if file_type == "txt":
+            output_file.write("")
+        elif file_type == "json":
```
Comment (nex-maximus) [inline temp.py:5]: Implement a more robust file type handling mechanism, potentially using MIME types or a dedicated library to support more file types in the future. [medium]
